packages/citizen-sdk/src/utils/triggerFaucet.ts

```diff
@@ -0,0 +1,227 @@
+// GoodSDKs/packages/citizen-sdk/src/utils/triggerFaucet.ts
+import type {
+  Address,
+  PublicClient,
+  WalletClient,
+  Chain,
+  Account,
+} from "viem"
+import { waitForTransactionReceipt } from "viem/actions"
+import { Envs, faucetABI, getGasPrice } from "../constants"
+
+export type TriggerFaucetResult =
+  | "skipped"               // throttled, not eligible, or balance sufficient
+  | "topped_via_contract"   // success via on-chain call
+  | "topped_via_api"        // success via backend fallback
+  | "error"                 // tried and failed
+
+export interface TriggerFaucetParams {
+  chainId: number
+  account: Address
+  publicClient: PublicClient
+  walletClient: WalletClient<any, Chain | undefined, Account | undefined>
+  faucetAddress: Address
+  env: string // "production" | "staging" | etc.
+  throttleMs?: number // default 1h
+}
+
+/**
+ * Checks if the user has sufficient balance to claim UBI.
+ * @param chainId - The chain ID
+ * @param account - User's wallet address
+ * @param publicClient - Viem public client
+ * @param faucetAddress - Faucet contract address
+ * @param toppingAmount - Amount the faucet tops up
+ * @param minTopping - Minimum topping percentage
+ * @returns True if the user can claim, false otherwise.
+ * @throws If gas price cannot be determined or balance check fails.
+ */
+async function canClaim(
+  chainId: number,
+  account: Address,
+  publicClient: PublicClient,
+  faucetAddress: Address,
+  toppingAmount: bigint,
+  minTopping: number
+): Promise<boolean> {
+  const gasPrice = getGasPrice(chainId)
+  if (!gasPrice) {
+    throw new Error(
+      "Cannot determine gasPrice for the current connected chain.",
+    )
+  }
+
+  const minBalance = (chainId === 42220 ? 250000n : 150000n) * gasPrice
+  const minThreshold =
+    (toppingAmount * (100n - BigInt(minTopping))) / 100n || minBalance
+
+  const balance = await publicClient.getBalance({
+    address: account,
+  })
+
+  return balance >= minThreshold
+}
+
+/**
+ * Contract-first faucet top-up with API fallback and throttling.
+ * 1) Check if user already has sufficient balance to claim
+ * 2) If eligible (canTop && balance < toppingAmount), try on-chain top-up (user signs).
+ * 3) Guard against gas > toppingAmount and insufficient gas to publish tx.
+ * 4) If on-chain path fails or cannot publish/sign, fallback to backend /verify/topWallet.
+ * 5) Throttle tops to once per chain per throttleMs via localStorage.
+ */
+export async function triggerFaucet({
+  chainId,
+  account,
+  publicClient, 
+  walletClient,
+  faucetAddress,
+  env,
+  throttleMs = 60 * 60 * 1000,
+}: TriggerFaucetParams): Promise<TriggerFaucetResult> {
+  // Throttle via localStorage (browser env only)
+  if (typeof localStorage !== "undefined") {
+    alert("TX needs to be signed in order to claim.")
+
+    const key = `goodDollarFaucetLastToppedUtcMs_${chainId}`
+    const last = localStorage.getItem(key)
```

Comment:
unncessary duplicate logic.

1. turn `last` variable in to let, move to top.
```
let last: string | null = null
```
2. invert if statement to 
```
(last && Date.now() > Number(last) + throttleMs) {
return "skipped"
}
```

remove everything else inside the localStorage !== undefined.
have the rest of the flow as is.